### PR TITLE
Fix rowMode='array' in postgres patcher

### LINKED ISSUE #364: postgres patcher breaks rowMode: array query
Querying of the form 
```js
c.query({
  text: <query>,
  rowMode: 'array'
})
```

gets passed to pg incorrectly without the rowMode param.  I tried to fix in #363 but may have broken something.

### PR DESCRIPTION
**Pull request by isaacl**
The patcher incorrectly strips rowMode from calls of the form:
c.query({ text: <query>, rowMode: 'array' })

Fixes #364.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by isaacl** [comment]
hmm patch is causing tests to fail, idk I'll file an issue

**Comment by isaacl** [comment]
I am still not sure this works, I had some weird behavior in xray but may be unrelated.

**Comment by willarmiros** [comment]
Hi @isaacl,

Thanks for this contribution, do you think you could add a unit test to verify this fixes the problematic behavior?

**Comment by isaacl** [comment]
OK I did a more involved patch where I use the argument logic from the pg module itself.

File: packages/postgres/test/unit/postgres_p.test.js [hunk 1]
```diff
@@ -231,4 +231,65 @@ describe('capturePostgres', function() {
...
+      sandbox.stub(segment, 'addNewSubsegment').returns(subsegment);
+      sandbox.stub(AWSXRay, 'isAutomaticMode').returns(true);
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should pass down raw sql query', function() {
+      query.call(postgres, 'sql here', ['values']);
```
Comment (willarmiros) [inline packages/postgres/test/unit/postgres_p.test.js:283]: Can we also assert that `addNewSubsegment` is `calledOnce` in both of these tests? Just to validate we are in fact correctly patching these queries in addition to propagating the params.

**Review by willarmiros — COMMENTED** [review]
Looks great to me! This will be an awesome simplification, just one small comment. Thanks!

File: packages/postgres/test/unit/postgres_p.test.js (hunk 1 above)
Comment (isaacl) [reply packages/postgres/test/unit/postgres_p.test.js:283]: sure, done

**Review by willarmiros — APPROVED** [review]
Thanks again! Verified with previous versions of the `pg` library that this change will be backwards compatible, since the `query()` API hasn't really changed apart from adding the query config object as an alternative to a string for the first argument, which this change adds support for.

**Comment by isaacl** [comment]
can haz release?

**Pull request merged**